Whiskey/Functions/Invoke-WhiskeyTask.ps1

```diff
@@ -178,105 +178,119 @@ function Invoke-WhiskeyTask
+            }
 
-        # PowerShell's default DebugPreference when someone uses the -Debug switch is `Inquire`. That would cause a build
-        # to hang, so let's set it to Continue so users can see debug output.
-        if( $parameter['Debug'] )
-        {
-            $DebugPreference = 'Continue'
-            $parameter.Remove('Debug')
-        }
+            $parameter = Get-TaskParameter -Name $task.CommandName -TaskProperty $taskProperties -Context $TaskContext
 
-        & $task.CommandName @parameter
-        $result = 'COMPLETED'
-    }
-    finally
-    {
-        $DebugPreference = $originalDebugPreference
+            # PowerShell's default DebugPreference when someone uses the -Debug switch is `Inquire`. That would cause a build
+            # to hang, so let's set it to Continue so users can see debug output.
+            if( $parameter['Debug'] )
+            {
+                $DebugPreference = 'Continue'
+                $parameter.Remove('Debug')
+            }
 
-        # Clean required tools *after* running the task since the task might need a required tool in order to do the cleaning (e.g. using Node to clean up installed modules)
-        if( $TaskContext.ShouldClean )
+            & $task.CommandName @parameter
+            $result = 'COMPLETED'
+        }
+        finally
         {
-            foreach( $requiredTool in $requiredTools )
+            $DebugPreference = $originalDebugPreference
+
+            # Clean required tools *after* running the task since the task might need a required tool in order to do the cleaning (e.g. using Node to clean up installed modules)
+            if( $TaskContext.ShouldClean )
             {
-                Uninstall-WhiskeyTool -BuildRoot $TaskContext.BuildRoot -ToolInfo $requiredTool
+                foreach( $requiredTool in $requiredTools )
+                {
+                    Uninstall-WhiskeyTool -BuildRoot $TaskContext.BuildRoot -ToolInfo $requiredTool
+                }
             }
-        }
 
-        if( $taskTempDirectory -and (Test-Path -Path $taskTempDirectory -PathType Container) )
-        {
-            Remove-Item -Path $taskTempDirectory -Recurse -Force -ErrorAction Ignore
+            if( $taskTempDirectory -and (Test-Path -Path $taskTempDirectory -PathType Container) )
+            {
+                Remove-Item -Path $taskTempDirectory -Recurse -Force -ErrorAction Ignore
+            }
+            $endedAt = Get-Date
+            $duration = $endedAt - $startedAt
+            Write-WhiskeyVerbose -Context $TaskContext -Message ('{0} in {1}' -f $result,$duration)
+            Write-WhiskeyVerbose -Context $TaskContext -Message ''
         }
-        $endedAt = Get-Date
-        $duration = $endedAt - $startedAt
-        Write-WhiskeyVerbose -Context $TaskContext -Message ('{0} in {1}' -f $result,$duration)
-        Write-WhiskeyVerbose -Context $TaskContext -Message ''
+
+        Invoke-Event -EventName 'AfterTask' -Property $taskProperties

```

Comment:
Two events?

Reply:
Users can register for a global `AfterTask` event, which fires after every task is run. They can also register an event to run after a specific task runs (e.g. `AfterCopyFile` task), just to help those who want to be as fast as possible.